### ISSUE TITLE
EnodeULRImplTest fails when run locally (on mac)

### ISSUE THREAD

**Issue by macfarla** [hyperledger/besu#4653]
```
expected: enode://6f8a80d14311c39f35f516fa664deaaaa13e85b2f7493f37f6144d86991ec012937307647bd3b9a82abe2974e1407241d54947bbb39763a4cac9f77166ad92a0@1100511:30303
 but was: enode://6f8a80d14311c39f35f516fa664deaaaa13e85b2f7493f37f6144d86991ec012937307647bd3b9a82abe2974e1407241d54947bbb39763a4cac9f77166ad92a0@0.16.202.223:30303
	at java.base@11.0.11/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base@11.0.11/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base@11.0.11/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at app//org.hyperledger.besu.ethereum.p2p.peers.EnodeURLImplTest.toURI_WithHostnameShouldWorkWhenDnsEnabledAndUpdateEnabled(EnodeURLImplTest.java:514)

``` 
java -version
openjdk version "11.0.16.1" 2022-08-12
OpenJDK Runtime Environment Homebrew (build 11.0.16.1+0)
OpenJDK 64-Bit Server VM Homebrew (build 11.0.16.1+0, mixed mode)

**Comment by Gabriel-Trintinalia** [comment]
For me: my hostname is a weird number and it resolves to an invalid (?) IP. If I change my hostname to
running: `sudo hostname localhost`  or `sudo hostname anything`, the test passes.

1200607 is my machine’s name. When I change it to something else it resolves to 127.0.0.1 and works fine. That may be the cause why it works for CircleCi and not locally.

maybe there is a bug in our logic, or the test is broken, or it is something specific to our laptops

**Comment by macfarla** [comment]
fixed by #4930
